### ISSUE TITLE
GPO Wish: Disable Clear History Manually from within options and Disable accessing Options at all

### ISSUE THREAD

**Issue by troysmith03** [mozilla/policy-templates#181]
I work for school district and now that FireFox finally is compatible with GPO's I am able to roll it out for student use. However, there are a couple things missing from the GPO templates that would be very useful in a school environment, First off Disabling the "Clear History" Button in Options and secondly Disabling access to "Options" completely. Students like to mess with settings and clear their history so they cant get in trouble when they find a way around the filter..... Both Chrome and IE support these lockouts. Please consider adding them to the new Firefox GPO template set

**Comment by jlpinzon** [comment]
I work in the non-profit world and regularly come across this of locking down history to prevent clearing it.  I'd like to tag on my request for this one since it's needed before my current organization will approve a massive rollout of Firefox.  Greatly appreciated. Thanks

**Comment by troysmith03** [comment]
yes please feel free ! it's one of the reasons I don't roll firefox out at many of my jh and hs buildings....

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by mkaply** [comment]
Does any other browser completely lock down deleting history?

When I look doing a policy like this, preventing access isn't enough. There are many places where you can delete individual entries from history. Do other browser do that?

**Comment by jlpinzon** [comment]
Thank you for your quick reply!  I'll rephrase: I understand that this block will only apply to the application and will not prevent the possibility of a knowledgeable user from deleting browsing history at the file system level.  The vast majority of our users at my organization will not know how to do that.  However, both Internet Explorer and Chrome allow the complete lock down of clearing history within the confines of their applications.  I just implemented this functionality in Chrome's GPO and it has explicit blocking of the clear browsing history option.

**Comment by mkaply** [comment]
Thanks, I'll take a look at what Chrome does.

**Comment by jlpinzon** [comment]
No problem, thanks!

**Comment by mkaply** [comment]
Just looked at Chrome and they do manage to remove most entry points. Although there is a bug that if you shift+delete on URL entries that are highlighted in the URL bar they still get deleted from history. Feel free to open a bug with Google on that one :)

**Comment by jlpinzon** [comment]
Wow good catch, thanks!  Think you could get Firefox to do the job better than Chrome did lol?

**Comment by jlpinzon** [comment]
I checked the bug.  I do see that it removes the site from the recent history list, however, I can still see it in the browsing history.  This is acceptable for my purposes.  I'm sure you can work your magic with Firefox and make it even better than Chrome.

**Comment by atlantafan** [comment]
Any update on this?  We would find this enhancement extremely valuable and necessary for us to move forward rolling Firefox out company wide.

**Comment by mkaply** [comment]
I was working on it, but it got sidetracked by other work. I'm definitely planning to have it in the next few months.

**Comment by emike09** [comment]
I'd love to have this feature as well. In our IT security audit, we were asked to prevent users from deleting history, but like others have said, we can't do this in FF yet. 

Thanks!

**Comment by ivan-volnov** [comment]
Any news here?

**Comment by mkaply** [comment]
I'm sorry, we started some work here but it didn't get finished. I'm really trying to get it done.

**Comment by mkaply** [comment]
Unfortunately it got stuck on lots of little things. STill on my radar.

**Comment by emike09** [comment]
We still block FF in our business because of the lack of this single ability. 3.04% global browser share, it could go up with this one simple change!

**Comment by kirwan-safesurfer** [comment]
@mkaply Just voicing my support for this feature too! I work for a parental controls company, and we would love to see a policy to disable just the clearing of history, like how [Chrome supports doing](https://chromeenterprise.google/policies/#AllowDeletingBrowserHistory).
